src/XrdHttp/XrdHttpProtocol.cc

```diff
@@ -1759,10 +1759,17 @@ void XrdHttpProtocol::Cleanup() {
 
   if (ssl) {
 
-
-    if (SSL_shutdown(ssl) != 1) {
-      TRACE(ALL, " SSL_shutdown failed");
-      ERR_print_errors(sslbio_err);
+    int ret = SSL_shutdown(ssl);
+    if (ret != 1) {
+        if(ret == 0) {
+            //https://www.openssl.org/docs/man1.0.2/man3/SSL_shutdown.html
+            //Call again SSL_shutdown
+            ret = SSL_shutdown(ssl);
```

Comment:
Is a single retry always sufficient? It seems that in [XrdTlsSocket it may be called up to 4 times](https://github.com/xrootd/xrootd/blob/master/src/XrdTls/XrdTlsSocket.cc#L752-L769) in order to shutdown SSL.

Reply:
Replying to my own comment here, but the documentation states that at most two calls are needed.